tests/test_profile.py

```diff
@@ -1,11 +1,68 @@
+from pathlib import Path
 from unittest import mock
 
-import pytest
-
 from nanopub import profile
 
 
-@mock.patch('nanopub.profile.get_profile', return_value={'orcid_id': ''})
-def test_no_orcid_id(mock_get_profile):
-    with pytest.raises(profile.ProfileError):
-        profile.get_orcid_id()
+def test_load_profile(tmpdir):
+    test_file = Path(tmpdir / 'profile.yml')
+    with test_file.open('w') as f:
+        f.write(
+                'orcid_id: pietje\n'
+                'name: https://orcid.org/0000-0000-0000-0000\n'
+                'public_key: /home/.nanopub/id_rsa.pub\n'
+                'private_key: /home/.nanopub/id_rsa\n')
+
+    with mock.patch('nanopub.profile.PROFILE_PATH', test_file):
+        profile.get_profile.cache_clear()
+        p = profile.get_profile()
+
+        assert p.orcid_id == 'pietje'
+        assert p.name == 'https://orcid.org/0000-0000-0000-0000'
+        assert p.public_key == Path('/home/.nanopub/id_rsa.pub')
+        assert p.private_key == Path('/home/.nanopub/id_rsa')
+        assert p.nanopub_uri is None
+
+
+def test_store_profile(tmpdir):
+    test_file = Path(tmpdir / 'profile.yml')
+
+    p = profile.Profile('pietje', 'https://orcid.org/0000-0000-0000-0000',
+                        Path('/home/.nanopub/id_rsa.pub'), Path('/home/.nanopub/id_rsa'))
+
+    with mock.patch('nanopub.profile.PROFILE_PATH', test_file):
+        profile.store_profile(p)
+
+        with test_file.open('r') as f:
+            assert f.read() == (
+                    'orcid_id: pietje\n'
+                    'name: https://orcid.org/0000-0000-0000-0000\n'
+                    'public_key: /home/.nanopub/id_rsa.pub\n'
+                    'private_key: /home/.nanopub/id_rsa\n')
+
+
+def test_nanopub_uri_roundtrip(tmpdir):
```

Comment:
What does this test exactly? How is it different than `test_load_profile` and `test_store_profile`?

Reply:
It's a little belt-and-suspenders, but it has `nanopub_uri` specified, which the other tests don't have.

Reply:
Ok, that seems overkill to me, but then again in the other comment I was saying we need more test coverage.... ;)

Reply:
Actually, there should probably be some negative tests which test various broken files and that they're handled correctly. The original code would raise a `KeyError` if the `public_key` attribute was missing, and a `FileNotFoundError` if the whole file was missing. It explicitly handles an empty file, by raising `ProfileError`. So ideally these would all be handled correctly (for whatever is decided to be correct), and tests would exist that would check that.

Reply:
Yeah, fair enough. I think we can be a bit less ambitious, but feel free to add more tests if you want to 😛

Reply:
I'll leave it at this then.